src/openforms/submissions/tasks/cleanup.py

```diff
@@ -60,24 +66,7 @@ def cleanup_on_completion_results():
         )
         with transaction.atomic():
             processing_status.forget_results()
-            submission.on_completion_task_ids = []
-            submission.save(update_fields=["on_completion_task_ids"])
-
-
-@app.task(ignore_result=True)
-def finalize_completion_retry(submission_id: int):
-    """
-    Finalize of the on_completion_retry workflow.
-
-    This task must be called last and mark the submission as not requiring any retries
-    anymore. Tasks being retried that fail again should result in a celery failure
-    to prevent this finalizer from running.
-
-    All other internal state will have been updated by the specific (sub)tasks.
-    """
-    submission = Submission.objects.get(id=submission_id)
-    submission.needs_on_completion_retry = False
-    submission.save(update_fields=["needs_on_completion_retry"])
+            submission.postcompletiontasksids_set.all().delete()
```

Comment:
I was concerned that we lose debug information, but apparently we already do in the current versions anyway. Since this considers only stuff that went right, it's probably okay to delete this data.

Reply:
Yes, I added it because of the line `submission.on_completion_task_ids = []` which was present before the refactor